openstack/blockstorage/v2/snapshots/requests.go

```diff
@@ -0,0 +1,158 @@
+package snapshots
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// CreateOptsBuilder allows extensions to add additional parameters to the
+// Create request.
+type CreateOptsBuilder interface {
+	ToSnapshotCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts contains options for creating a Snapshot. This object is passed to
+// the snapshots.Create function. For more information about these parameters,
+// see the Snapshot object.
+type CreateOpts struct {
+	VolumeID    string                 `json:"volume_id" required:"true"`
+	Force       bool                   `json:"force,omitempty"`
+	Name        string                 `json:"name,omitempty"`
+	Description string                 `json:"description,omitempty"`
+	Metadata    map[string]interface{} `json:"metadata,omitempty"`
+}
+
+// ToSnapshotCreateMap assembles a request body based on the contents of a
+// CreateOpts.
+func (opts CreateOpts) ToSnapshotCreateMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "snapshot")
+}
+
+// Create will create a new Snapshot based on the values in CreateOpts. To
+// extract the Snapshot object from the response, call the Extract method on the
+// CreateResult.
+func Create(client *gophercloud.ServiceClient, opts CreateOptsBuilder) (r CreateResult) {
+	b, err := opts.ToSnapshotCreateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	_, r.Err = client.Post(createURL(client), b, &r.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{200, 201},
```

Comment:
I did a quick copy of the [v1 snapshot acceptance tests](https://github.com/gophercloud/gophercloud/tree/master/acceptance/openstack/blockstorage/v1) to test v2 and I got the following error:

```
=== RUN   TestSnapshotsList
--- PASS: TestSnapshotsList (0.29s)
=== RUN   TestSnapshotsCreateDelete
--- FAIL: TestSnapshotsCreateDelete (5.46s)
        blockstorage.go:53: Attempting to create volume: ACPTTESTIY8ygDIM6NjjZdWr
        blockstorage.go:25: Attempting to create snapshot ACPTTESTEPU7UzMfgx0Z1lD1 based on volume 72f8d398-d682-4bd1-b3f3-b8ecc32a387a
        snapshots_test.go:48: Unable to create snapshot: Expected HTTP response code [200 201] when accessing [POST https://foo:8776/v2/9aa5f9f66b4b417d84d778a23acdf45b/snapshots], but got 202 instead
                {"snapshot": {"status": "creating", "description": null, "created_at": "2017-06-01T04:38:54.860173", "metadata": {}, "volume_id": "72f8d398-d682-4bd1-b3f3-b8ecc32a387a", "size": 1, "id": "29be3eb4-738c-4db1-bad6-8d18082dbb45", "name": "ACPTTESTEPU7UzMfgx0Z1lD1"}}
        blockstorage.go:138: Unable to delete volume 72f8d398-d682-4bd1-b3f3-b8ecc32a387a: Invalid request due to incorrect syntax or missing required parameters.
FAIL
```

I updated the `OkCodes` with just `202` (to match v2 volumes) and the tests passed. Is 202 the right code to use here (and below)?